refactor: Product card action template

aragon999 opened the pull request:
### 1. Why is this change necessary?
Currently the blocks in the `product/card/action.html.twig` template are not named consistently. Furthermore there exists an block with two inputs, outside of an form, which are used for the google analytics plugin. One is already present in the actual product form. But to be compatible with the detail button, the information is now added as data attribute to the product box itself.

### 2. What does this change do, exactly?
Rename blocks, deprecate some blocks, cleanup the action template and handling of the `ViewItemListEvent`.

### 3. Describe each step to reproduce the issue or behaviour.
\-

### 4. Please link to the relevant issues (if any).
\-

### 5. Checklist

- [x] I have rebased my changes to remove merge conflicts
- [ ] I have written tests and verified that they fail without my change
- [x] I have created a [changelog file](https://github.com/shopware/platform/blob/trunk/adr/2020-08-03-implement-new-changelog.md) with all necessary information about my changes
- [ ] I have written or adjusted the documentation according to my changes
- [x] This change has comments for package types, values, functions, and non-obvious lines of code
- [x] I have read the contribution requirements and fulfil them.

aragon999 committed:
refactor: Product card action template

tobiasberge commented on src/Storefront/Resources/views/storefront/component/product/card/action.html.twig:
```diff
@@ -72,20 +82,25 @@
-        <input type="hidden"
-               name="product-name"
-               value="{{ product.translated.name }}">
+    {# @deprecated tag:v6.7.0 - Block, including the content will be removed without replacement #}
```
We can also mention that the data is now inside `data-product-information`.

```suggestion
    {# @deprecated tag:v6.7.0 - Block, including the content will be removed. "product-name" and "product-id" will be hold inside "data-product-information" in box-standard.html.twig instead. #}
```

tobiasberge commented on src/Storefront/Resources/views/storefront/component/product/card/action.html.twig:
```diff
@@ -28,40 +27,51 @@
-                            </div>
+                        {% block component_product_box_action_buy_button %}
+                            {# @deprecated tag:v6.7.0 - Block will be removed, use component_product_box_action_buy_button instead #}
+                            {% block page_product_detail_product_buy_button %}
```
Good catch with the block names. 👍

tobiasberge commented on src/Storefront/Resources/app/storefront/src/plugin/google-analytics/events/view-item-list.event.js:
```diff
@@ -27,22 +27,28 @@ export default class ViewItemListEvent extends AnalyticsEvent
-
-            if (!id || !name) {
-                return;
+            if (window.Feature.isActive('v6.7.0.0')) {
```
```suggestion
            /** @deprecated tag:v6.7.0 - Data will be retrieved from the data attribute "data-product-information" instead of hidden inputs. */
            if (window.Feature.isActive('v6.7.0.0')) {
```

tobiasberge reviewed (commented):
Hi @aragon999,

thank you for your contribution 🎉 and sorry for my late review.

LGTM so far. 👍 I would agree to not have those arbitrary hidden inputs outside the form when this is only used by Analytics.

I just had some small suggestions about the deprecation comments. If it's ok for you we can also apply them when we import the PR.

aragon999 commented:
hey @tobiasberge no problem :-) 

The adjustments of the deprecation commments seem reasonable, I just rebased the commit and adjusted the deprecation comments :-)

ghost commented:
Hello,

thank you for creating this pull request.
I have opened an issue on our Issue Tracker for you. See the issue link: https://issues.shopware.com/issues/NEXT-34415

Please use this issue to track the state of your pull request.

tobiasberge commented:
Hi @aragon999,

your changes were merged to trunk https://github.com/shopware/shopware/commit/d1639f09abbb6563f2bd635e62fb480376afbe40

Thank you for your contribution. 🎉
